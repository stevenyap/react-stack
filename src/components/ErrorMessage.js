/**
 * @flow
 * Responsibility: Renders an error alert message
 */
import React from 'react'
import { Alert } from 'react-bootstrap/lib'

type Props = { text: string }
const ErrorMessage = (props: Props) => {
  const { text } = props

  return (
    <Alert bsStyle="danger">
      <p>{text}</p>
    </Alert>
  )
}

export default ErrorMessage
