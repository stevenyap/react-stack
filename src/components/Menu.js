/**
 * @flow
 * Responsibility: Display the menu
 */
import React from 'react'
import { Link } from 'react-router-dom'

export const Menu = () => {
  return (
    <ul>
      <li><Link to="/">Home</Link></li>
    </ul>
  )
}

export default Menu
