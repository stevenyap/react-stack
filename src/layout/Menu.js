/**
 * @flow
 * Responsibility: Display the menu
 */
import type { ContextRouter } from 'react-router-dom'

import React from 'react'
import { connect } from 'react-redux'
import { Nav, Navbar, NavItem } from 'react-bootstrap/lib'
import { Link, withRouter } from 'react-router-dom'

import { doLogout } from 'ducks/admin'

type MenuLinkProps = { to: string, text: string } & ContextRouter
const MenuLink = withRouter((props: MenuLinkProps) => {
  const { to, text, history } = props
  const handleClick = () => history.push(to)
  return <NavItem href="#" onClick={handleClick}>{text}</NavItem>
})

type ComponentState = { admin: Admin }
type Props = ComponentState & { ispatch: Dispatch }
export const Menu = (props: Props) => {
  const { admin, dispatch } = props
  const logout = () => dispatch(doLogout())

  return (
    <Navbar inverse={true} collapseOnSelect={true}>
      <Navbar.Header>
        <Navbar.Brand>
          <Link to="/">React Stack</Link>
        </Navbar.Brand>
        {admin && <Navbar.Toggle />}
      </Navbar.Header>
      {admin &&
        <Navbar.Collapse>
          <Nav pullRight>
            <MenuLink to="/" text="Home" />
            <NavItem href="#" onClick={logout}>Logout</NavItem>
          </Nav>
        </Navbar.Collapse>}
    </Navbar>
  )
}

const mapStateToProps = (state: State): ComponentState => ({
  admin: state.admin.data
})
export default connect(mapStateToProps)(Menu)
