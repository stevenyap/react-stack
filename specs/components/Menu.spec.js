import React from 'react'
import { BrowserRouter } from 'react-router-dom'

import { Menu } from 'components/Menu'

describe('Menu', () => {
  it('renders correctly', () => {
    const tree = renderJSON(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()

    const renderedText = toText(tree)
    expect(renderedText).toContain('Home')
  })
})
