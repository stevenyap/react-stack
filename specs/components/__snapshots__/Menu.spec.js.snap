// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Logo
      </a>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav navbar-right"
        role={null}
      >
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            Home
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
